After 0.26, cmake configuration has fail with 'Unable identify compiler architecture for CMAKE_SYSTEM_PROCESSOR'

I think changes affects to my project is [this](https://github.com/aminya/project_options/pull/168).
My project is fine until v0.25.2.

I use visual studio 2022 17.4.4 with integrated cmake, and pre-installed vcpkg.
My cmake file is mostly similar with [example file](https://github.com/aminya/cpp_vcpkg_project/blob/main/CMakeLists.txt).
Difference is that I don't use `run_vcpkg()` functionality.
Instead, I register 'CMAKE_TOOL_CHAIN' using CMakePresets.json, just like visual studio's [official manual](https://learn.microsoft.com/en-us/cpp/build/cmake-presets-vs?view=msvc-170#enable-vcpkg-integration).

I'm not so sure about that it is appropriate to report here or '[WindowsToolchain](https://github.com/MarkSchofield/WindowsToolchain)' cause it is dependency not 'project_option' itself.
but I thought that it is something worth to report.

```
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_TOOLCHAIN_FILE:FILEPATH="C:\dev\vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_C_COMPILER:STRING="cl" -DCMAKE_CXX_COMPILER:STRING="cl" -DCMAKE_BUILD_TYPE:STRING="Debug" -DFEATURE_UNIT_TESTS:STRING="ON" -DCMAKE_INSTALL_PREFIX:PATH="E:/work/Knuckles/Game/out/install/windows-msvc-debug-soft-mode"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\work\Knuckles\Game" 2>&1"
1> Working directory: E:/work/Knuckles/Game/out/build/windows-msvc-debug-soft-mode
1> [CMake] CMake Warning (dev) at C:/Program Files/Microsoft Visual Studio/2022/Professional/Common7/IDE/CommonExtensions/Microsoft/CMake/CMake/share/cmake-3.24/Modules/FetchContent.cmake:1264 (message):
1> [CMake]   The DOWNLOAD_EXTRACT_TIMESTAMP option was not given and policy CMP0135 is
1> [CMake]   not set.  The policy's OLD behavior will be used.  When using a URL
1> [CMake]   download, the timestamps of extracted files should preferably be that of
1> [CMake]   the time of extraction, otherwise code that depends on the extracted
1> [CMake]   contents might not be rebuilt if the URL changes.  The OLD behavior
1> [CMake]   preserves the timestamps from the archive instead, but this is usually not
1> [CMake]   what you want.  Update your project to the NEW behavior or specify the
1> [CMake]   DOWNLOAD_EXTRACT_TIMESTAMP option with a value of true to avoid this
1> [CMake]   robustness issue.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeLists.txt:24 (FetchContent_Declare)
1> [CMake] This warning is for project developers.  Use -Wno-dev to suppress it.
1> [CMake] 
1> [CMake] -- Using Windows Windows toolchain
1> [CMake] CMake Error at out/build/windows-msvc-debug-soft-mode/_deps/_msvc_toolchain-src/Windows.MSVC.toolchain.cmake:169 (message):
1> [CMake]   Unable identify compiler architecture for CMAKE_SYSTEM_PROCESSOR
1> [CMake] Call Stack (most recent call first):
1> [CMake]   out/build/windows-msvc-debug-soft-mode/_deps/_project_options-src/src/VCEnvironment.cmake:72 (include)
1> [CMake]   out/build/windows-msvc-debug-soft-mode/_deps/_project_options-src/src/Index.cmake:34 (msvc_toolchain)
1> [CMake]   out/build/windows-msvc-debug-soft-mode/_deps/_project_options-src/CMakeLists.txt:3 (include)
1> [CMake] -- Configuring incomplete, errors occurred!
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_TOOLCHAIN_FILE:FILEPATH="C:\dev\vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_C_COMPILER:STRING="cl" -DCMAKE_CXX_COMPILER:STRING="cl" -DCMAKE_BUILD_TYPE:STRING="Debug" -DFEATURE_UNIT_TESTS:STRING="ON" -DCMAKE_INSTALL_PREFIX:PATH="E:/work/Knuckles/Game/out/install/windows-msvc-debug-soft-mode"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\work\Knuckles\Game" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_TOOLCHAIN_FILE:FILEPATH="C:\dev\vcpkg/scripts/buildsystems/vcpkg.cmake" -DCMAKE_C_COMPILER:STRING="cl" -DCMAKE_CXX_COMPILER:STRING="cl" -DCMAKE_BUILD_TYPE:STRING="Debug" -DFEATURE_UNIT_TESTS:STRING="ON" -DCMAKE_INSTALL_PREFIX:PATH="E:/work/Knuckles/Game/out/install/windows-msvc-debug-soft-mode"  -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\PROFESSIONAL\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "E:\work\Knuckles\Game" 2>&1"' returned with exit code: 1'.
```

Issue reproduced today with 0.31 on Windows 
```
- Using Windows Windows toolchain
CMake Error at cmake-build-release/_deps/_msvc_toolchain-src/Windows.MSVC.toolchain.cmake:166 (message):
  **Unable identify compiler architecture for CMAKE_SYSTEM_PROCESSOR**
Call Stack (most recent call first):
  cmake-build-release/_deps/_project_options-src/src/VCEnvironment.cmake:65 (include)
  cmake-build-release/_deps/_project_options-src/src/Index.cmake:38 (msvc_toolchain)
  cmake-build-release/_deps/_project_options-src/CMakeLists.txt:3 (include)
```

while version 0.25.2 correctly set this variable :

```
-- Running vcpkg install - done
BUILD TYPE : Release USING :MSVC version : 19.36.32537.0
CMAKE_SYSTEM_PROCESSOR = x64
CMAKE_SYSTEM_NAME = Windows
CMAKE_SYSTEM_VERSION = 
CMAKE_SYSTEM = Windows
CMAKE_CXX_COMPILER = C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe
CMAKE_CXX_COMPILER_VERSION = 19.36.32537.0
CMAKE_CXX_COMPILER_ID = MSVC
CMAKE_CXX_COMPILER_FRONTEND_VARIANT = MSVC
CMAKE_CXX_COMPILER_TARGET = 
CMAKE_CXX_COMPILER_EXTERNAL_TOOLCHAIN = 
CMAKE_CXX_COMPILER_LAUNCHER = 
CMAKE_CXX_COMPILER_WORKS = TRUE
```

--- a/src/Index.cmake
+++ b/src/Index.cmake
@@ -35,7 +35,11 @@ include("${CMAKE_CURRENT_LIST_DIR}/DynamicProjectOptions.cmake")
 include("${CMAKE_CURRENT_LIST_DIR}/Hardening.cmake")
 
 # Include msvc toolchain on windows if the generator is not visual studio. Should be called before run_vcpkg and run_conan to be effective
-msvc_toolchain()
+if("${CMAKE_TOOL_CHAIN}" STREQUAL "")
+  msvc_toolchain()
+else()
+  message(STATUS "project_options: skipping msvc_toolchain")
+endif()
 
 include("${CMAKE_CURRENT_LIST_DIR}/Conan.cmake")
 include("${CMAKE_CURRENT_LIST_DIR}/Vcpkg.cmake")

